framework/caching/CacheInterface.php

```diff
@@ -0,0 +1,224 @@
+     */
+    public function flush();
+
+    /**
+     * Returns whether there is a cache entry with a specified key.
+     * This method is required by the interface [[\ArrayAccess]].
+     * @param string $key a key identifying the cached value
+     * @return bool
+     */
+    public function offsetExists($key);
+
+    /**
+     * Retrieves the value from cache with a specified key.
+     * This method is required by the interface [[\ArrayAccess]].
+     * @param string $key a key identifying the cached value
+     * @return mixed the value stored in cache, false if the value is not in the cache or expired.
+     */
+    public function offsetGet($key);
+
+    /**
+     * Stores the value identified by a key into cache.
+     * If the cache already contains such a key, the existing value will be
+     * replaced with the new ones. To add expiration and dependencies, use the [[set()]] method.
+     * This method is required by the interface [[\ArrayAccess]].
+     * @param string $key the key identifying the value to be cached
+     * @param mixed $value the value to be cached
+     */
+    public function offsetSet($key, $value);
+
+    /**
+     * Deletes the value with the specified key from cache
+     * This method is required by the interface [[\ArrayAccess]].
+     * @param string $key the key of the value to be deleted
+     */
+    public function offsetUnset($key);
+
+    /**
+     * Method combines both [[set()]] and [[get()]] methods to retrieve value identified by a $key,
+     * or to store the result of $callable execution if there is no cache available for the $key.
+     *
+     * Usage example:
+     *
+     * ```php
+     * public function getTopProducts($count = 10) {
+     *     $cache = $this->cache; // Could be Yii::$app->cache
+     *     return $cache->getOrSet(['top-n-products', 'n' => $count], function ($cache) use ($count) {
+     *         return Products::find()->mostPopular()->limit(10)->all();
+     *     }, 1000);
+     * }
+     * ```
+     *
+     * @param mixed $key a key identifying the value to be cached. This can be a simple string or
+     * a complex data structure consisting of factors representing the key.
+     * @param callable|\Closure $callable the callable or closure that will be used to generate a value to be cached.
+     * In case $callable returns `false`, the value will not be cached.
+     * @param int $duration default duration in seconds before the cache will expire. If not set,
+     * [[defaultDuration]] value will be used.
+     * @param Dependency $dependency dependency of the cached item. If the dependency changes,
+     * the corresponding value in the cache will be invalidated when it is fetched via [[get()]].
+     * This parameter is ignored if [[serializer]] is `false`.
+     * @return mixed result of $callable execution
+     * @since 2.0.11
+     */
+    public function getOrSet($key, $callable, $duration = null, $dependency = null);
```

Comment:
Same as `buildKey()`- I've added it to the interface because the interface API should be compatible with current abstract class Cache.

Otherwise we should write everywhere in PHPDocs something like `CacheInterface|Cache`.
We can move these methods to a separate `ExtendedCacheInterface` or so. @yiisoft/core-developers